generator/routing_index_generator.cpp

```diff
@@ -78,25 +120,87 @@ class Processor final
     }
   }
 
-  VehicleMask CalcVehicleMask(FeatureType const & f) const
-  {
-    VehicleMask mask = 0;
-    if (m_pedestrianModel->IsRoad(f))
-      mask |= kPedestrianMask;
-    if (m_bicycleModel->IsRoad(f))
-      mask |= kBicycleMask;
-    if (m_carModel->IsRoad(f))
-      mask |= kCarMask;
-
-    return mask;
-  }
-
-  shared_ptr<IVehicleModel> const m_pedestrianModel;
-  shared_ptr<IVehicleModel> const m_bicycleModel;
-  shared_ptr<IVehicleModel> const m_carModel;
+  VehicleMaskMaker const m_maskMaker;
   unordered_map<uint64_t, Joint> m_posToJoint;
   unordered_map<uint32_t, VehicleMask> m_masks;
 };
+
+bool BordersContains(vector<m2::RegionD> const & borders, m2::PointD const & point)
+{
+  for (m2::RegionD const & region : borders)
+  {
+    if (region.Contains(point))
+      return true;
+  }
+
+  return false;
+}
+
+void CalcCrossMwmTransitions(string const & path, string const & mwmFile, string const & country,
+                             vector<CrossMwmRampSerializer::Transition> & transitions,
+                             vector<CrossMwmRamp> & ramps)
+{
+  string const polyFile = my::JoinFoldersToPath({path, BORDERS_DIR}, country + BORDERS_EXTENSION);
```

Comment:
Could you please rewrite JoinFoldersToPath to variable templates and get rid of this ugly curly braces?